### PR TITLE
Update most char submenu popups to new popup

### PR DESCRIPTION
**Pull request by Wolfsblvt**
- Update 'Scenario Override'
- Update 'Open Source'
- Update 'Replace Character'
- Update 'Convert to Persona' popups

## Checklist:

- [x] I have read the [Contributing guidelines](https://www.youtube.com/watch?v=dQw4w9WgXcQ).

### CODE REVIEW

File: public/script.js [hunk 1]
```diff
@@ -7184,20 +7184,23 @@ function updateFavButtonState(state) {
...
-    template.find('[data-group="true"]').toggle(isGroup);
-    template.find('[data-character="true"]').toggle(!isGroup);
-    template.find('.chat_scenario').val(metadataValue).on('input', onScenarioOverrideInput);
-    template.find('.remove_scenario_override').on('click', onScenarioOverrideRemoveClick);
-    callPopup(template, 'text');
+
+    const $template = $(await renderTemplateAsync('scenarioOverride'));
+    $template.find('[data-group="true"]').toggle(isGroup);
+    $template.find('[data-character="true"]').toggle(!isGroup);
+    // TODO: Why does this save on every character input? Save on popup close
```
Comment (Cohee1207) [inline public/script.js:7199]: Shouldn't be much of a deal since it uses debounced metadata save

File: public/script.js [hunk 2]
```diff
@@ -10674,7 +10677,7 @@ jQuery(async function () {
                 const source = getCharacterSource(this_chid);
                 if (source && isValidUrl(source)) {
                     const url = new URL(source);
-                    const confirm = await callPopup(`Open ${url.hostname} in a new tab?`, 'confirm');
+                    const confirm = await Popup.show.confirm('Open Source', `<span>Do you want to open the link to ${url.hostname} in a new tab?</span><var>${url}</var>`);
```
Comment (Cohee1207) [inline public/script.js:10680]: No DOMPurify?

Comment (Cohee1207) [reply public/script.js:10680]: Never mind.

**Review by Cohee1207 — APPROVED** [review]

File: public/script.js (hunk 1 above)
Comment (Wolfsblvt) [reply public/script.js:7199]: It feels wrong to have popups that handle input while typing/working with them, and not on close.
But yeah, it's not a real issue here. That's where I left it in.

**Pull request merged**